### ISSUE TITLE
NOT_FOUND_COLUMN_IN_BLOCK  in aggregation from a subquery for some functions with a where clause

### ISSUE THREAD

**Issue by kashwy** [ClickHouse/ClickHouse#42771]
> You have to provide the following information whenever possible.

```
 create table test 
(
    "name" String,
     "time" Int64 ) 
ENGINE=MergeTree() primary key (time) order by (time);

insert into test Values('hello world', 1662336000241);
```
run query:
```
SELECT count()
FROM
(
    SELECT
        fromUnixTimestamp64Milli(time, 'UTC') AS time_fmt,
        name
    FROM test
    WHERE time_fmt > '2022-09-05 00:00:00'
);
```
or 
```
SELECT count() FROM ( SELECT toInt128(time) AS time_fmt, name FROM test WHERE time_fmt > 1662336000240)
```
get the following error:
```

0 rows in set. Elapsed: 0.010 sec.

Received exception from server (version 22.10.1):
Code: 10. DB::Exception: Received from localhost:9000. DB::Exception: Not found column time in block. (NOT_FOUND_COLUMN_IN_BLOCK)
```

if no where  clause.
the result is correct:
```
SELECT count() FROM ( SELECT toInt128(time) AS time_fmt, name FROM test)

┌─count()─┐
│       1 │
└─────────┘
```

**Does it reproduce on recent release?**
yes
[The list of releases](https://github.com/ClickHouse/ClickHouse/blob/master/utils/list-versions/version_date.tsv)

**How to reproduce**
run the script above 
* Which ClickHouse server version to use
version 22.10.1

**Expected behavior**

should get the result of the count 

```
count()|
-------+
      1|
```

**Additional context**

but there is no problem in the version 22.8.1.726

**Comment by alexey-milovidov** [comment]
> but there is no problem in the version 22.8.1.726

No, it is broken even in 22.8.1: https://fiddle.clickhouse.com/0fcdadf2-7b55-4409-a2b8-af6d7a389db0

Broken between 22.7 and 22.8.

**Comment by alexey-milovidov** [comment]
@amosbird is it related to index analysis?

**Comment by alexey-milovidov** [comment]
Reproduced in master:

```
milovidov-desktop :) create table test 
                     (
                         "name" String,
                          "time" Int64 ) 
                     ENGINE=MergeTree() primary key (time) order by (time);

CREATE TABLE test
(
    `name` String,
    `time` Int64
)
ENGINE = MergeTree
PRIMARY KEY time
ORDER BY time

Query id: b4d9ee3a-deba-437d-9640-ac00bc41fd47

Ok.

0 rows in set. Elapsed: 0.021 sec. 

milovidov-desktop :) insert into test Values('hello world', 1662336000241);
                     

INSERT INTO test FORMAT Values

Query id: df96bfd8-6fc1-4007-b129-2cf928c5e3a3

Ok.

1 row in set. Elapsed: 0.002 sec. 

milovidov-desktop :) SELECT count()
                     FROM
                     (
                         SELECT
                             fromUnixTimestamp64Milli(time, 'UTC') AS time_fmt,
                             name
                         FROM test
                         WHERE time_fmt > '2022-09-05 00:00:00'
                     );
                     

SELECT count()
FROM
(
    SELECT
        fromUnixTimestamp64Milli(time, 'UTC') AS time_fmt,
        name
    FROM test
    WHERE time_fmt > '2022-09-05 00:00:00'
)

Query id: 9555be9e-6267-4a4e-b217-a21efc13ef97


0 rows in set. Elapsed: 0.002 sec. 

Received exception:
Code: 10. DB::Exception: Not found column time in block. (NOT_FOUND_COLUMN_IN_BLOCK)
```

**Comment by larryluogit** [comment]
@alexey-milovidov Could you trigger back port this fix to 22.8 please? we are waiting for this fix.
